client-common/src/main/java/org/apache/cassandra/sidecar/common/http/SidecarHttpHeaderNames.java

```diff
@@ -31,4 +31,8 @@ public final class SidecarHttpHeaderNames
      * {@code "cassandra-content-xxhash32-seed"}
      */
     public static final String CONTENT_XXHASH32_SEED = "cassandra-content-xxhash32-seed";
+    /**
+     * {@code "cassandra-auth-role"}
+     */
+    public static final String AUTH_ROLE = "cassandra-auth-role";
```

Comment:
It would worth a detailed explanation on the implication of this header and how it is to be used, to answer questions like what if a malicious user set a unexpected role (with high permission) on purpose, would it be a security hole.